diff --git a/includes/webform.components.inc b/includes/webform.components.inc
index eb4f17a..8e569f4 100644
--- a/includes/webform.components.inc
+++ b/includes/webform.components.inc
@@ -938,6 +938,7 @@ function webform_component_feature($type, $feature) {
     'group' => FALSE,
     'attachment' => FALSE,
     'private' => TRUE,
+    'placeholder' => FALSE,
     'wrapper_classes' => TRUE,
     'css_classes' => TRUE,
     'views_range' => FALSE,
diff --git a/webform.module b/webform.module
index 09e4cbd..554fe13 100644
--- a/webform.module
+++ b/webform.module
@@ -935,6 +935,7 @@ function webform_webform_component_info() {
       'features' => array(
         'email_address' => TRUE,
         'spam_analysis' => TRUE,
+        'placeholder' => TRUE,
       ),
     ),
     'fieldset' => array(
@@ -1037,6 +1038,7 @@ function webform_webform_component_info() {
       'file' => 'components/textarea.inc',
       'features' => array(
         'spam_analysis' => TRUE,
+        'placeholder' => TRUE,
       ),
     ),
     'textfield' => array(
@@ -1046,6 +1048,7 @@ function webform_webform_component_info() {
       'features' => array(
         'email_name' => TRUE,
         'spam_analysis' => TRUE,
+        'placeholder' => TRUE,
       ),
     ),
     'time' => array(
@@ -2684,6 +2687,12 @@ function _webform_client_form_add_component($node, $component, $component_value,
         $element['#webform_private'] = $component['extra']['private'];
       }
 
+      // The 'placeholder' option is in some components, but it's not a real
+      // property. Add it for Form Builder compatibility.
+      if (webform_component_feature($component['type'], 'placeholder')) {
+        $element['#webform_placeholder'] = $component['extra']['placeholder'];
+      }
+
       // Add custom CSS classes to the field and wrapper.
       _webform_component_classes($element, $component);
 
@@ -4579,3 +4588,4 @@ function _webform_submission_serial_next_value($nid, $do_increment = TRUE) {
 
   return $next_serial;
 }
+
