diff --git a/inc/form.inc b/inc/form.inc
index 607ebef..4794a3d 100644
--- a/inc/form.inc
+++ b/inc/form.inc
@@ -67,15 +67,21 @@ function zurb_foundation_form_alter(&$form, &$form_state, $form_id) {
  * Implements theme_form_element_label().
  */
 function zurb_foundation_form_element_label($variables) {
- _zurb_foundation_add_tooltips($variables);
- return theme_form_element_label($variables);
+  // Add tooltips to form elements.
+  if (theme_get_setting('zurb_foundation_tooltip_enable')) {
+    _zurb_foundation_add_tooltips($variables);
+  }
+  return theme_form_element_label($variables);
 }
 
 /**
  * Implements theme_text_format_wrapper().
  */
 function zurb_foundation_text_format_wrapper($variables) {
-  _zurb_foundation_add_tooltips($variables);
+  // Add tooltips to form elements.
+  if (theme_get_setting('zurb_foundation_tooltip_enable')) {
+    _zurb_foundation_add_tooltips($variables);
+  }
   return theme_text_format_wrapper($variables);
 }
 
@@ -83,7 +89,10 @@ function zurb_foundation_text_format_wrapper($variables) {
  * Implements theme_field_multiple_value_form().
  */
 function zurb_foundation_field_multiple_value_form($variables) {
-  _zurb_foundation_add_tooltips($variables);
+  // Add tooltips to form elements.
+  if (theme_get_setting('zurb_foundation_tooltip_enable')) {
+    _zurb_foundation_add_tooltips($variables);
+  }
   return theme_field_multiple_value_form($variables);
 }
 
