diff --git a/crumbs.module b/crumbs.module
index 7fca86f..3bd7672 100644
--- a/crumbs.module
+++ b/crumbs.module
@@ -151,6 +151,17 @@ function crumbs_menu() {
     );
   }
 
+  if (module_exists('menu_firstchild')) {
+    $items['crumbs/firstchild-menu-item/%menu_link'] = array(
+      'access callback' => TRUE,
+      'page callback' => 'crumbs_firstchild_menu_link_page',
+      'page arguments' => array(2),
+      'title callback' => 'crumbs_special_menu_link_title',
+      'title arguments' => array(2),
+      'type' => MENU_CALLBACK,
+    );
+  }
+
   return $items;
 }
 
@@ -164,6 +175,13 @@ function crumbs_special_menu_link_page(array $menu_link) {
 /**
  * @param array $menu_link
  */
+function crumbs_firstchild_menu_link_page(array $menu_link) {
+  drupal_goto('<front>');
+}
+
+/**
+ * @param array $menu_link
+ */
 function crumbs_special_menu_link_title(array $menu_link) {
   return $menu_link['title'];
 }
@@ -627,4 +645,4 @@ function _crumbs_admin_flush_cache() {
  */
 function _crumbs_unserialize_failure($class) {
   throw new crumbs_UnserializeException();
-}
\ No newline at end of file
+}
diff --git a/lib/Router.php b/lib/Router.php
index 21edc2c..f9e8e00 100644
--- a/lib/Router.php
+++ b/lib/Router.php
@@ -50,6 +50,9 @@ class crumbs_Router {
     if ('crumbs_special_menu_link_page' === $item['page_callback']) {
       $item['href'] = '<nolink>';
     }
+    else if('crumbs_firstchild_menu_link_page' === $item['page_callback']) {
+      $item['href'] = isset($item['fragments'][2]) ? _menu_firstchild_get_firstchild_href($item['fragments'][2]) : '<nolink>';
+    }
 
     if ($normalpath !== $item['href']) {
       $pos = strlen($item['href']);
diff --git a/plugins/crumbs.menu.inc b/plugins/crumbs.menu.inc
index b1a7eac..1709c72 100644
--- a/plugins/crumbs.menu.inc
+++ b/plugins/crumbs.menu.inc
@@ -52,6 +52,9 @@ class menu_CrumbsMultiPlugin_hierarchy implements crumbs_MultiPlugin_FindParentI
         if ('<nolink>' === $row->parent_path) {
           $candidates[$row->menu_name] = 'crumbs/special-menu-item/' . $row->plid;
         }
+        else if('<firstchild>' === $row->parent_path) {
+          $candidates[$row->menu_name] = 'crumbs/firstchild-menu-item/' . $row->plid;
+        }
         else {
           $candidates[$row->menu_name] = $row->parent_path;
         }
@@ -193,6 +196,10 @@ class menu_CrumbsMonoPlugin_specialItems implements crumbs_MonoPlugin_FindParent
     if ('<nolink>' === $parent_path) {
       return 'crumbs/special-menu-item/' . $menu_link['plid'];
     }
+    
+    if('<firstchild>' === $parent_path) {
+      return 'crumbs/firstchild-menu-item/' . $menu_link['plid'];
+    }
 
     if ('<separator>' === $parent_path) {
       return;
